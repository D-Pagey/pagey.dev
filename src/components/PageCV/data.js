import memrise from './assets/memrise.png';
import deliveroo from './assets/deliveroo.png';
import cytora from './assets/cytora.svg';
import drover from './assets/drover.svg';
import clearscore from './assets/clearscore.svg';
import titan from './assets/titan.png';

export default [
    {
        dates: `May '21 - Present`,
        description: ['I joined as the 3rd full time employee to build the core product.'],
        height: '75px',
        imageAlt: 'titanAcademy',
        image: titan,
        jobTitle: 'Lead Front End Engineer',
        technologies: ['react', 'typescript', 'styled components', 'jest', 'react-testing-library'],
        website: 'https://www.tracktitan.io/',
        width: 'auto'
    },
    {
        dates: `July '20 - April '21`,
        description: [
            'I joined as the 4th developer in the Web team to rebuild the learning sessions from a Python/Django app into React and TypeScript. The overall goal was to allow the web experience to gain feature parity with the native mobile experience, which had previously been a priority.',
            'I also fundamentally changed how the tech team reviewed code, changing the priority to be about shipping frequently and often.'
        ],
        height: '75px',
        imageAlt: 'memrise',
        image: memrise,
        jobTitle: 'Software Engineer',
        technologies: ['react', 'typescript', 'styled components', 'jest', 'react-testing-library'],
        website: 'https://www.memrise.com/',
        width: 'auto'
    },
    {
        dates: `Jan '20 - July '20`,
        description: [
            `I joined the Consumer Pricing team as a Software Engineer, our team works on anything to do with costs to our customers. This involves doing pieces of work in a variety of Deliveroo products and quickly getting to grips with the nuances of each team's project.`,
            `I've recently shipped a large project that allows our restaurant partners to set unique fees for different postcodes. I was the sole Front End Developer, and I was also involved in the ideation process, user-testing, managing the project prioritisation and workload within the team.`,
            `Outside of my team's project work, I'm involved in: implementing Cypress as an end-to-end testing library, advocating for better testing strategies, regularly contributing to our multiple component libraries and design systems, and addressing tech debt where appropriate.`
        ],
        height: 'auto',
        imageAlt: 'deliveroo',
        image: deliveroo,
        jobTitle: 'Software Engineer',
        technologies: ['react', 'typescript', 'redux', 'jest', 'enzyme', 'css-modules', 'circle ci'],
        website: 'https://deliveroo.co.uk',
        width: '200px'
    },
    {
        dates: `May '19 - Jan '20`,
        description: [
            'I was hired to Cytora as the first Front End Developer to build out UIs from scratch on top of the API platform. In addition, I created a reusable component library to be able to build future UIs quickly and consistently.',
            'I wrote and deployed Node cloud functions on Google Cloud Platform to abstract and aggregate API calls from the Front End. This empowered my team to release new features without relying on other teams to amend their microservices.'
        ],
        imageAlt: 'cytora',
        image: cytora,
        jobTitle: 'Front End Engineer',
        technologies: [
            'react',
            'node',
            'react-context',
            'jest',
            'react-testing-library',
            'styled components',
            'storybook',
            'cypress',
            'circle ci',
            'concourse',
            'gcp'
        ],
        website: 'https://cytora.com/'
    },
    {
        dates: `Sept '18 - May '19`,
        description: [
            'I joined Drover as a Junior Front End Developer and my first project was working solo on rebuilding Ruby on Rails pages into React. Whilst I was confident working in React, I quickly learnt how to work in a large codebase with lots of shared components (and minimal tests!).',
            'After 6 months I was promoted to a mid-level Developer position. I took advantage of working in a small team by suggesting and implementing improvements quickly. For example, I was the flag bearer for doing more integration and end-to-end testing for critical user flows.'
        ],
        imageAlt: 'drover',
        image: drover,
        jobTitle: 'Front End Developer',
        technologies: [
            'react',
            'redux',
            'jest',
            'enzyme',
            'styled components',
            'storybook',
            'cypress',
            'heroku',
            'aws'
        ],
        website: 'https://www.joindrover.com/'
    },
    {
        dates: `May '17 - Sept '18`,
        description: [
            'I joined ClearScore as roughly employee #100 and, by the time I left, the company had grown to around 200 employees internationally.',
            'Beyond my day-to-day recruitment role, I created and ran a company-wide interview training programme that focused on bringing all interviewers on to the same page. This aimed to ensure the interview process was fair and unbiased, provided candidates with useful feedback and raised awareness around unconscious bias.',
            'I also made multiple contributions to the front-end production code base which varied from updating integrations on the careers page to bug fixes on the home page.'
        ],
        imageAlt: 'clearscore',
        image: clearscore,
        jobTitle: 'Talent Acquisition',
        technologies: ['es6', 'handlebars', 'css', 'jenkins'],
        website: 'https://clearscore.com/'
    }
];
