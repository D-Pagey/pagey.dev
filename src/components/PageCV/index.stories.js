import React from 'react';
import { PageCV } from '.';

export default { title: 'PageCV' };

export const normal = () => <PageCV />;
