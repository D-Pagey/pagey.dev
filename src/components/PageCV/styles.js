import styled from 'styled-components';

export const Wrapper = styled.div`
    margin: 0 auto;
    max-width: 600px;
`;

export const Title = styled.h1`
    font-family: 'Merriweather', serif;
`;

export const List = styled.ul`
    list-style: none;
    padding: 0;
`;

export const Item = styled.li`
    margin: 2rem 0 0;
`;

export const FlexWrapper = styled.div`
    align-items: center;
    display: flex;
    justify-content: space-between;
    margin: 1rem 0 0;
`;

export const CompanyLink = styled.a`
    cursor: pointer;
`;

export const Logo = styled.img`
    display: block;
    margin: 0 auto;
    height: 50px;
    width: 150px;
`;

export const JobTitle = styled.h5`
    font-weight: 600;
    font-size: 1.125rem;
    margin: 0;
`;

export const ItalicSpan = styled.span`
    font-style: italic;
`;

export const LinkedInLinkWrapper = styled.div`
    align-items: center;
    display: flex;
    flex-direction: column;
    margin: 2rem 0 1rem;
`;

export const Link = styled.a`
    margin: 0.5rem 0 0;

    @media (min-width: 500px) {
        margin-top: 1rem;
    }
`;

export const Image = styled.img`
    width: 125px;
`;
