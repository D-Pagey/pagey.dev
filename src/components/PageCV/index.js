import React from 'react';
import cytora from './assets/cytora.svg';
import drover from './assets/drover.svg';
import clearscore from './assets/clearscore.svg';
import * as S from './styles';
import Pills from '../Pills';

const PageCV = () => (
    <S.Wrapper data-testid="pageCV">
        <S.Title>Mini CV</S.Title>

        <S.SubTitle>Dan Page</S.SubTitle>

        <p>
            I am a Software Developer that loves building applications in JavaScript. I’m always thinking about coding,
            whether that is what I want to learn next or what I’m going to build next.{' '}
        </p>

        <S.List>
            <S.Item>
                <S.CompanyLink href="https://cytora.com/" target="_blank">
                    <S.Logo src={cytora} alt="cytora" />
                </S.CompanyLink>
                <S.FlexWrapper>
                    <S.JobTitle>Front End Engineer</S.JobTitle>
                    <S.ItalicSpan>May &apos;19 - Present</S.ItalicSpan>
                </S.FlexWrapper>

                <p>
                    I joined as the first Front End Developer to build out UIs from scratch on top of the API platform.
                    I&apos;m also creating a reusable component library to enable multiple teams to build UIs quickly
                    and consistently.
                </p>

                <Pills
                    items={[
                        'react',
                        'react-context',
                        'styled components',
                        'storybook',
                        'jest',
                        'react-testing-library',
                        'cypress',
                        'circle ci',
                        'concourse',
                        'gcp'
                    ]}
                />
            </S.Item>

            <S.Item>
                <S.CompanyLink href="https://www.joindrover.com/" target="_blank">
                    <S.Logo src={drover} alt="drover" />
                </S.CompanyLink>
                <S.FlexWrapper>
                    <S.JobTitle>Front End Developer</S.JobTitle>
                    <S.ItalicSpan>Sept &apos;18 - May &apos;19</S.ItalicSpan>
                </S.FlexWrapper>
                <p>
                    I joined Drover as a Junior Front End Developer and my first project was working on rebuilding Ruby
                    on Rails pages in React. The main learning curve came from using shared components that had no unit
                    or integration tests, which made refactoring challenging.
                </p>

                <p>
                    After 6 months I was promoted to a mid-level Developer and put on a 3 month greenfield project to
                    rebuild the entire user booking flow. I was the flag bearer for doing more unit and integration
                    testing and so I spent a lot of my free time learning about Jest, Enzyme, React-Testing-Library and
                    Cypress.
                </p>

                <Pills
                    colour="pink"
                    items={[
                        'react',
                        'redux',
                        'styled components',
                        'storybook',
                        'jest',
                        'enzyme',
                        'cypress',
                        'heroku',
                        'aws'
                    ]}
                />
            </S.Item>

            <S.Item>
                <S.CompanyLink href="https://clearscore.com/" target="_blank">
                    <S.Logo src={clearscore} alt="clearscore" />
                </S.CompanyLink>
                <S.FlexWrapper>
                    <S.JobTitle>Talent Acquisition</S.JobTitle>
                    <S.ItalicSpan>May &apos;17 - Sept &apos;18</S.ItalicSpan>
                </S.FlexWrapper>

                <p>
                    I joined ClearScore as roughly employee #100 and, by the time I left, the company had grown to
                    around 200 employees internationally.
                </p>

                <p>
                    Beyond my day-to-day recruitment role, I created and ran a company-wide interview training programme
                    that focused on bringing all interviewers on to the same page. This aimed to ensure the interview
                    process was fair and unbiased, provided candidates with useful feedback and raised awareness around
                    unconscious bias.
                </p>

                <p>
                    I also made multiple contributions to the front-end production code base which varied from updating
                    integrations on the careers page to bug fixes on the home page.
                </p>

                <Pills colour="lightgreen" items={['es6', 'handlebars', 'css', 'jenkins']} />
            </S.Item>
        </S.List>
    </S.Wrapper>
);

export default PageCV;
