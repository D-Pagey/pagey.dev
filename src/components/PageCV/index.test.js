import React from 'react';
import PageCV from '.';

describe('PageCV component', () => {
    it('should render', () => {
        const { container } = render(<PageCV />);
        expect(container.firstChild).toMatchSnapshot();
    });
});
