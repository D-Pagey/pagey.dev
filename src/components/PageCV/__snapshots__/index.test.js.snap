// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageCV component should render 1`] = `
.c3 {
  margin: 2rem 0 0;
}

.c6 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  margin: 1rem 0 0;
}

.c4 {
  cursor: pointer;
}

.c5 {
  display: block;
  margin: 0 auto;
  height: 75px;
  width: auto;
}

.c7 {
  font-weight: 600;
  font-size: 1.125rem;
  margin: 0;
}

.c8 {
  font-style: italic;
}

.c9 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  list-style: none;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0;
}

.c10 {
  background-color: #fbce63;
  border-radius: 5px;
  font-size: 12px;
  padding: 0.25rem 0.5rem;
  margin: 0 0 0.5rem 0.5rem;
}

.c0 {
  margin: 0 auto;
  max-width: 600px;
}

.c1 {
  font-family: 'Merriweather',serif;
}

.c2 {
  list-style: none;
  padding: 0;
}

.c12 {
  font-style: italic;
}

.c11 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  margin: 2rem 0 1rem;
}

.c13 {
  margin: 0.5rem 0 0;
}

.c14 {
  width: 125px;
}

@media (min-width:500px) {
  .c13 {
    margin-top: 1rem;
  }
}

<div
  class="c0"
  data-testid="pageCV"
>
  <h1
    class="c1"
  >
    Dan Page
  </h1>
  <p>
    I am a Software Developer that loves building applications in JavaScript. I’m always thinking about coding, whether that is what I want to learn next or what I’m going to build next.
     
  </p>
  <ul
    class="c2"
  >
    <li
      class="c3"
    >
      <a
        class="c4"
        href="https://www.memrise.com/"
        target="_blank"
      >
        <img
          alt="memrise"
          class="c5"
          height="75px"
          src="memrise.png"
          width="auto"
        />
      </a>
      <div
        class="c6"
      >
        <h5
          class="c7"
        >
          Software Engineer
        </h5>
        <span
          class="c8"
        >
          July '20 - Present
        </span>
      </div>
      <div>
        <p>
          I have joined the Web team to help the web experience have feature parity with the native mobile experience.
        </p>
      </div>
      <ul
        class="c9"
      >
        <li
          class="c10"
        >
          react
        </li>
        <li
          class="c10"
        >
          typescript
        </li>
        <li
          class="c10"
        >
          styled components
        </li>
        <li
          class="c10"
        >
          jest
        </li>
        <li
          class="c10"
        >
          react-testing-library
        </li>
      </ul>
    </li>
    <li
      class="c3"
    >
      <a
        class="c4"
        href="https://deliveroo.co.uk"
        target="_blank"
      >
        <img
          alt="deliveroo"
          class="c5"
          height="75px"
          src="deliveroo.png"
          width="auto"
        />
      </a>
      <div
        class="c6"
      >
        <h5
          class="c7"
        >
          Software Engineer
        </h5>
        <span
          class="c8"
        >
          Jan '20 - July '20
        </span>
      </div>
      <div>
        <p>
          I joined the Consumer Pricing team as a Software Engineer, our team works on anything to do with costs to our customers. This involves doing pieces of work in a variety of Deliveroo products and quickly getting to grips with the nuances of each team's project.
        </p>
        <p>
          I've recently shipped a large project that allows our restaurant partners to set unique fees for different postcodes. I was the sole Front End Developer, and I was also involved in the ideation process, user-testing, managing the project prioritisation and workload within the team.
        </p>
        <p>
          Outside of my team's project work, I'm involved in: implementing Cypress as an end-to-end testing library, advocating for better testing strategies, regularly contributing to our multiple component libraries and design systems, and addressing tech debt where appropriate.
        </p>
      </div>
      <ul
        class="c9"
      >
        <li
          class="c10"
        >
          react
        </li>
        <li
          class="c10"
        >
          typescript
        </li>
        <li
          class="c10"
        >
          redux
        </li>
        <li
          class="c10"
        >
          jest
        </li>
        <li
          class="c10"
        >
          enzyme
        </li>
        <li
          class="c10"
        >
          css-modules
        </li>
        <li
          class="c10"
        >
          circle ci
        </li>
      </ul>
    </li>
    <li
      class="c3"
    >
      <a
        class="c4"
        href="https://cytora.com/"
        target="_blank"
      >
        <img
          alt="cytora"
          class="c5"
          height="75px"
          src="cytora.svg"
          width="auto"
        />
      </a>
      <div
        class="c6"
      >
        <h5
          class="c7"
        >
          Front End Engineer
        </h5>
        <span
          class="c8"
        >
          May '19 - Jan '20
        </span>
      </div>
      <div>
        <p>
          I was hired to Cytora as the first Front End Developer to build out UIs from scratch on top of the API platform. In addition, I created a reusable component library to be able to build future UIs quickly and consistently.
        </p>
        <p>
          I wrote and deployed Node cloud functions on Google Cloud Platform to abstract and aggregate API calls from the Front End. This empowered my team to release new features without relying on other teams to amend their microservices.
        </p>
      </div>
      <ul
        class="c9"
      >
        <li
          class="c10"
        >
          react
        </li>
        <li
          class="c10"
        >
          node
        </li>
        <li
          class="c10"
        >
          react-context
        </li>
        <li
          class="c10"
        >
          jest
        </li>
        <li
          class="c10"
        >
          react-testing-library
        </li>
        <li
          class="c10"
        >
          styled components
        </li>
        <li
          class="c10"
        >
          storybook
        </li>
        <li
          class="c10"
        >
          cypress
        </li>
        <li
          class="c10"
        >
          circle ci
        </li>
        <li
          class="c10"
        >
          concourse
        </li>
        <li
          class="c10"
        >
          gcp
        </li>
      </ul>
    </li>
    <li
      class="c3"
    >
      <a
        class="c4"
        href="https://www.joindrover.com/"
        target="_blank"
      >
        <img
          alt="drover"
          class="c5"
          height="75px"
          src="drover.svg"
          width="auto"
        />
      </a>
      <div
        class="c6"
      >
        <h5
          class="c7"
        >
          Front End Developer
        </h5>
        <span
          class="c8"
        >
          Sept '18 - May '19
        </span>
      </div>
      <div>
        <p>
          I joined Drover as a Junior Front End Developer and my first project was working solo on rebuilding Ruby on Rails pages into React. Whilst I was confident working in React, I quickly learnt how to work in a large codebase with lots of shared components (and minimal tests!).
        </p>
        <p>
          After 6 months I was promoted to a mid-level Developer position. I took advantage of working in a small team by suggesting and implementing improvements quickly. For example, I was the flag bearer for doing more integration and end-to-end testing for critical user flows.
        </p>
      </div>
      <ul
        class="c9"
      >
        <li
          class="c10"
        >
          react
        </li>
        <li
          class="c10"
        >
          redux
        </li>
        <li
          class="c10"
        >
          jest
        </li>
        <li
          class="c10"
        >
          enzyme
        </li>
        <li
          class="c10"
        >
          styled components
        </li>
        <li
          class="c10"
        >
          storybook
        </li>
        <li
          class="c10"
        >
          cypress
        </li>
        <li
          class="c10"
        >
          heroku
        </li>
        <li
          class="c10"
        >
          aws
        </li>
      </ul>
    </li>
    <li
      class="c3"
    >
      <a
        class="c4"
        href="https://clearscore.com/"
        target="_blank"
      >
        <img
          alt="clearscore"
          class="c5"
          height="75px"
          src="clearscore.png"
          width="auto"
        />
      </a>
      <div
        class="c6"
      >
        <h5
          class="c7"
        >
          Talent Acquisition
        </h5>
        <span
          class="c8"
        >
          May '17 - Sept '18
        </span>
      </div>
      <div>
        <p>
          I joined ClearScore as roughly employee #100 and, by the time I left, the company had grown to around 200 employees internationally.
        </p>
        <p>
          Beyond my day-to-day recruitment role, I created and ran a company-wide interview training programme that focused on bringing all interviewers on to the same page. This aimed to ensure the interview process was fair and unbiased, provided candidates with useful feedback and raised awareness around unconscious bias.
        </p>
        <p>
          I also made multiple contributions to the front-end production code base which varied from updating integrations on the careers page to bug fixes on the home page.
        </p>
      </div>
      <ul
        class="c9"
      >
        <li
          class="c10"
        >
          es6
        </li>
        <li
          class="c10"
        >
          handlebars
        </li>
        <li
          class="c10"
        >
          css
        </li>
        <li
          class="c10"
        >
          jenkins
        </li>
      </ul>
    </li>
  </ul>
  <div
    class="c11"
  >
    <span
      class="c12"
    >
      For further career history, check out my profile on: 
    </span>
    <a
      class="c13"
      href="https://www.linkedin.com/in/dan-page/"
      rel="noopener noreferrer"
      target="_blank"
    >
      <img
        alt="linkedin"
        class="c14"
        src="LI-Logo.png"
      />
    </a>
  </div>
</div>
`;
